### ISSUE TITLE
Accessibility: Home page audience tabs

### ISSUE THREAD

**Issue by banksashleym** [JacksonCollege/jc-theme-2015#271]
In continuing my testing using the keyboard, I'm noticing that the audience tabs on the home page "future students, current students, etc..." are hard to navigate using a keyboard. When I get to "Future students" and click on enter nothing happens, and when I press on tab again the focus moves to the next tab instead of the content within the tab.

To remedy this, when the focus is one one of the main tabs and they press enter, could the tab then focus on the links inside the tab box? so for instance tab to future students>enter>tab would move to the future students link inside the tab> enter and that would take them to the future students page....then if there are no other links the tab would move to the next link "current students" and then each tab would take them through the links in that box "internships, current students, employment services.." 

Thanks!

**Comment by d4mation** [comment]
This change is on Staging 👍 

I set it up so that if they hit "Enter" while focusing a Tab, it will take them to the first focusable item within the Tab's Panel. If there are no focusable Items within a Tab's Panel, then it does not adjust the focus, so hitting "Tab" again puts focus on the next Tab instead of continuing down the page from within the Tab's Panel.

**Comment by banksashleym** [comment]
This looks great too! I did notice that when I tab to future students and hit enter it will tab to future students, which is perfect. But after that secondary future student link it then jumps to "Jackson College News" Is there a way that after a user has tabbed through those secondary links to have the focus then jump back up to the audience tabs and finish focusing on those?

**Comment by d4mation** [comment]
I just pushed a fix for this to Staging.

The way that it works now, if they press "Enter" to enter into a Tab's Panel and reach the end of the focusable items within it, it will take the focus to the next Tab (Provided there is a next Tab to go to. If it was the last tab, it will continue down the page). If they were focused on the first focusable item in the Tab Panel and pressed "Shift+Tab", it will take focus back up to the Tab that they were in.

Normally a "Tab Trap" is bad, but I think in this case it is a good thing as it will make traversing those Tabs much easier 👍

**Comment by d4mation** [comment]
Although, I did notice just now that I accidentally made a loop if you're just trying to loop through the Tabs quickly without entering any of them. I'll let you know when I've pushed a fix for that 👍

**Comment by d4mation** [comment]
What we could possibly do instead would be to not require hitting "Enter" and just expand the Tab Panel whenever the Tab comes into Focus. That would allow you to Tab through each Tab and Tab Panel in succession similarly to how the Navigation Menu is currently working on Staging. It should preserve how it works for Mouse/Touchscreen as-is as well.

**Comment by banksashleym** [comment]
Let's do that - I love that idea. I think it makes sense to keep it similar to the main navigation. I would much rather a person have to tab through and know those links are there instead of getting caught in a tab trap or somehow miss those links altogether. Thank you!

**Comment by d4mation** [comment]
Just pushed that change up 👍 

Should be able to tab forward and backward properly through those while auto-expanding the Tabs and moving through their contents.

**Comment by banksashleym** [comment]
This is great!!

**Comment by d4mation** [comment]
This change has been pushed to Live 👍
